net/ip_cache.go

```diff
@@ -1,9 +1,13 @@
 package net
 
-import "sync"
+import (
+	"sync"
+
+	"github.com/google/btree"
+)
 
 const (
-	ipCacheInitialSize = 1000000
+	ipCacheBTreeGrade = 3500
```

Comment:
why 3500?

Reply:
Because it's very good!
Seriously: I've benchmarked this to be fastest on my machine :D